src/actions/write/tick.ts

```diff
@@ -0,0 +1,78 @@
+import { DEFAULT_UNDERNAME_COUNT, SECONDS_IN_A_YEAR } from '../../constants';
+import { ContractResult, IOState } from '../../types';
+
+declare const SmartWeave: any;
+
+// Removes gateway from the gateway address registry after the leave period completes
+export const tick = async (state: IOState): Promise<ContractResult> => {
+  const { records, auctions, reserved } = state;
+
+  // TODO: do we give the caller a tip?
+
+  // remove any expired records
```

Comment:
Is this necessary? I can see the utility of being able to view an expired record, and any functions that look at it should be able to tell if its expired, eg auctions and buy record.

Reply:
eh - you can load the contract at specific heights if you're looking to mine that information.